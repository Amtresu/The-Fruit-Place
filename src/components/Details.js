import React from 'react';
import { useSelector } from 'react-redux';
import LaunchDetail from './Detail';

function Details() {
    let launches = useSelector((state) => state.launch)
    launches =launches.filter(launch => launch.toggle)
    return (
        <div>
            {launches.map((launch)=> (
                <LaunchDetail 
                key={launch.id}
                name={launch.name}
                id={launch.id}
                year={launch.year}
                upcoming={launch.upcoming}
                rocket_name = {launch.rocket_name}
                launch_success = {launch.launch_success}
                rocket_type = {launch.rocket_type}
                launch_site = {launch.launch_site}
                launch_date = {launch.launch_date}
                image = {launch.image}
                    />
            ))}
        </div>
    )
}

export default Details