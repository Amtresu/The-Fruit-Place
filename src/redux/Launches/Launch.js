/* eslint-disable */
import { createAsyncThunk } from '@reduxjs/toolkit';
const API = 'https://api.spacexdata.com/v3/launches'
const initialState = []
const GET_LAUNCHES = 'GET_LAUNCHES'
const SHOW_DETAILS ='SHOW_DETAILS'
const header = { 'Content-Type': 'application/json', 'Access-Control-Allow-Origin': '*',}

export const showDetails = (id) => ({
  type: SHOW_DETAILS,
  payload: id,
});

const getLaunches = createAsyncThunk(GET_LAUNCHES, async () => {
    const data = await fetch(API, {
      method: 'GET',
      headers: { 'Content-Type': 'application/json' }
    }).then((response) => response.json());
    let item;
    try {
      const api = await data;
      item = api.map((launch) => ({
        key: launch.launch_date_unix,
        id: launch.launch_date_unix,
        name: launch.mission_name,
        year: launch.launch_year,
        launch_date: launch.launch_date_utc,
        rocket_name: launch.rocket.rocket_name,
        rocket_type: launch.rocket.rocket_type,
        payloads: launch.rocket,
        launch_site: launch.launch_site.site_name_long,
        launch_success: launch.launch_success,
        details: launch.details,
        image: launch.links.mission_patch_small,
        upcoming: false,
      }));
    } catch (error) {
      console.error('ERR', error);
    }
    return item;
  });

  export default (state = initialState, action) => {
    switch (action.type) {
      case `${GET_LAUNCHES}/fulfilled`:
        if(state == 0 )
        console.log(action.payload.slice(0,30))
        return action.payload.slice(0,30)
        case SHOW_DETAILS:
      return state.map((launch) => {
        if (launch.id !== action.payload) {
          return launch;
        }
        console.log(state.filter(x => !!x))
        return  {...launch, toggle: !launch.toggle}
      }); 
      default:
        return state;
    }
  };
  export { getLaunches };
