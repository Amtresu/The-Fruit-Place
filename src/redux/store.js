import {
    combineReducers,
    configureStore,
    applyMiddleware,
  } from '@reduxjs/toolkit';
  import thunk from 'redux-thunk';
import launchReducer from './Launches/Launch'


  const rootReducer = combineReducers({
    launch: launchReducer,
  });
  
  const store = configureStore({ reducer: rootReducer }, applyMiddleware(thunk));
  
  export default store;
