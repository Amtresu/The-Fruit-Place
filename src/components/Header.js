import React from "react";
import Fruits from "./Launches";
import '../stylesheets/style.css'
import mic from '../images/mic.png'
import settings from '../images/settings.png'
import arrow from '../images/down-arrow.png'
import logo from '../images/spacex-logo-png-transparent.png'
import Launches from "./Launches";

const Header = () => {
    return (
        <div className="header-div">
            <header>
                <div className="header-nav">
                    <div>
                        <img src={arrow} alt='back-arrow' className="arrow icon" />
                    </div>
                    <div>
                        <img src={mic} alt="mic" className="mic icon" />
                        <img src={settings} alt="settings icon" className="settings" />
                    </div>
                </div>
                <div className="logo-div">
                    <img src={logo} alt="logo"className="logo"/>
                    <h1>The Launch Place</h1>
                </div>
            </header>
            <div className="">
                <Launches />
            </div>
        </div>
    )
}

export default Header

