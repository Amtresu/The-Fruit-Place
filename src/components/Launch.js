import React from "react";
import { useDispatch } from 'react-redux';
import { showDetails } from "../redux/Launches/Launch";
import { Link } from "react-router-dom";
import arrow from '../images/next-icon.svg'

const Launch = (props) => {
    const dispatch = useDispatch();
    const { id, name, year, image } = props

    const showDetails1 = () => {
        dispatch(showDetails(id));
    };
    return (
        <div key={id} className="mission-div">
             <Link
                id={id}
                type="button"
                onClick={showDetails1}
                to="./Details"
            ><img src={arrow} alt='back-arrow' className="home-arrow icon" /> </Link>
            <img src={image} alt='mission patch' className="mission-img"></img>
            <ul className="mission-list">
            <li>{name}</li>
            <li>{year}</li>
            </ul>
        </div>
    )
}

export default Launch