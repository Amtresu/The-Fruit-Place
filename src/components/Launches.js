import { React, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { getLaunches } from '../redux/Launches/Launch';
import Launch from './Launch';

function Launches() {
    const [query, setQuery] = useState("")
    const dispatch = useDispatch()
    const fruits = useSelector((state) => state.launch)

    useEffect(() => {
        dispatch(getLaunches())
    }, [])

    return (
        <div>
            <div className='input-div'>
        <input type='text' placeholder='Search. . .' onChange={event => setQuery(event.target.value)} />
        </div>
        <div className='fruits-div'>
            {
                fruits.filter(launch => {
                    if (query === '') {
                        return launch;
                    } else if (launch.name.toLowerCase().includes(query.toLowerCase())) {
                        return launch;
                    }
                }).map((launch, index) => (
                    <div className="box" key={index}>
                        <Launch
                            key={launch.id}
                            name={launch.name}
                            id={launch.id}
                            year={launch.year}
                            image = {launch.image}

                        />
                    </div>
                ))}
        </div>
        </div>
                )}

export default Launches