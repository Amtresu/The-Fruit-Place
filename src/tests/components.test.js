/* eslint-disable */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Header from '../components/Header';
import Details from '../components/Details';
import Launch from '../components/Launch';
import Launches from '../components/Launches';


describe('Render All Components', () => {
    test('render Header', () => {

        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Header />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <div className="header-div">
            <header>
                <div className="header-nav">
                    <div>
                        <img src={arrow} alt='back-arrow' className="arrow icon" />
                    </div>
                    <div>
                        <img src={mic} alt="mic" className="mic icon" />
                        <img src={settings} alt="settings icon" className="settings" />
                    </div>
                </div>
                <div className="logo-div">
                    <img src={logo} alt="logo"className="logo"/>
                    <h1>The Launch Place</h1>
                </div>
            </header>
            <div className="">
                <Launches />
            </div>
        </div>`);
    });
    test('render Launches', () => {

        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Launches />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
            <div>
            <div className='input-div'>
        <input type='text' placeholder='Search. . .' onChange={event => setQuery(event.target.value)} />
        </div>
        <div className='fruits-div'>
            {
                fruits.filter(launch => {
                    if (query === '') {
                        return launch;
                    } else if (launch.name.toLowerCase().includes(query.toLowerCase())) {
                        return launch;
                    }
                }).map((launch, index) => (
                    <div className="box" key={index}>
                        <Launch
                            key={launch.id}
                            name={launch.name}
                            id={launch.id}
                            year={launch.year}
                            image = {launch.image}

                        />
                    </div>
                ))}
        </div>
        </div>
                )
)`
        );
    });
    test('render Launch', () => {

        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Launch />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
    <div key={id} className="mission-div">
    <Link
       id={id}
       type="button"
       onClick={showDetails1}
       to="./Details"
   ><img src={arrow} alt='back-arrow' className="home-arrow icon" /> </Link>
   <img src={image} alt='mission patch' className="mission-img"></img>
   <ul className="mission-list">
   <li>{name}</li>
   <li>{year}</li>
   </ul>
</div>`
        );
    });
    test('render detail', () => {

        const { container } = render(
            <MemoryRouter>
                <Provider store={store}>
                    <Details />
                </Provider>
            </MemoryRouter>,
        );
        expect(container.firstChild).toMatchSnapshot(`
        <div>
            {launches.map((launch)=> (
                <LaunchDetail 
                key={launch.id}
                name={launch.name}
                id={launch.id}
                year={launch.year}
                upcoming={launch.upcoming}
                rocket_name = {launch.rocket_name}
                launch_success = {launch.launch_success}
                rocket_type = {launch.rocket_type}
                launch_site = {launch.launch_site}
                launch_date = {launch.launch_date}
                image = {launch.image}
                    />
            ))}
        </div>`
        );
    });
})