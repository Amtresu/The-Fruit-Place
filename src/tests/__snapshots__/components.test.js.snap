// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render All Components render Header: 
        <div className="header-div">
            <header>
                <div className="header-nav">
                    <div>
                        <img src={arrow} alt='back-arrow' className="arrow icon" />
                    </div>
                    <div>
                        <img src={mic} alt="mic" className="mic icon" />
                        <img src={settings} alt="settings icon" className="settings" />
                    </div>
                </div>
                <div className="logo-div">
                    <img src={logo} alt="logo"className="logo"/>
                    <h1>The Launch Place</h1>
                </div>
            </header>
            <div className="">
                <Launches />
            </div>
        </div> 1`] = `
<div
  class="header-div"
>
  <header>
    <div
      class="header-nav"
    >
      <div>
        <img
          alt="back-arrow"
          class="arrow icon"
          src="down-arrow.png"
        />
      </div>
      <div>
        <img
          alt="mic"
          class="mic icon"
          src="mic.png"
        />
        <img
          alt="settings icon"
          class="settings"
          src="settings.png"
        />
      </div>
    </div>
    <div
      class="logo-div"
    >
      <img
        alt="logo"
        class="logo"
        src="spacex-logo-png-transparent.png"
      />
      <h1>
        The Launch Place
      </h1>
    </div>
  </header>
  <div
    class=""
  >
    <div>
      <div
        class="input-div"
      >
        <input
          placeholder="Search. . ."
          type="text"
        />
      </div>
      <div
        class="fruits-div"
      />
    </div>
  </div>
</div>
`;

exports[`Render All Components render Launch: 
    <div key={id} className="mission-div">
    <Link
       id={id}
       type="button"
       onClick={showDetails1}
       to="./Details"
   ><img src={arrow} alt='back-arrow' className="home-arrow icon" /> </Link>
   <img src={image} alt='mission patch' className="mission-img"></img>
   <ul className="mission-list">
   <li>{name}</li>
   <li>{year}</li>
   </ul>
</div> 1`] = `
<div
  class="mission-div"
>
  <a
    href="/Details"
    type="button"
  >
    <img
      alt="back-arrow"
      class="home-arrow icon"
      src="down-arrow.png"
    />
     
  </a>
  <img
    alt="mission patch"
    class="mission-img"
  />
  <ul
    class="mission-list"
  >
    <li />
    <li />
  </ul>
</div>
`;

exports[`Render All Components render Launches: 
            <div>
            <div className='input-div'>
        <input type='text' placeholder='Search. . .' onChange={event => setQuery(event.target.value)} />
        </div>
        <div className='fruits-div'>
            {
                fruits.filter(launch => {
                    if (query === '') {
                        return launch;
                    } else if (launch.name.toLowerCase().includes(query.toLowerCase())) {
                        return launch;
                    }
                }).map((launch, index) => (
                    <div className="box" key={index}>
                        <Launch
                            key={launch.id}
                            name={launch.name}
                            id={launch.id}
                            year={launch.year}
                            image = {launch.image}

                        />
                    </div>
                ))}
        </div>
        </div>
                )
) 1`] = `
<div>
  <div
    class="input-div"
  >
    <input
      placeholder="Search. . ."
      type="text"
    />
  </div>
  <div
    class="fruits-div"
  />
</div>
`;

exports[`Render All Components render detail: 
        <div>
            {launches.map((launch)=> (
                <LaunchDetail 
                key={launch.id}
                name={launch.name}
                id={launch.id}
                year={launch.year}
                upcoming={launch.upcoming}
                rocket_name = {launch.rocket_name}
                launch_success = {launch.launch_success}
                rocket_type = {launch.rocket_type}
                launch_site = {launch.launch_site}
                launch_date = {launch.launch_date}
                image = {launch.image}
                    />
            ))}
        </div> 1`] = `<div />`;
